libmscore/tuplet.cpp

```diff
@@ -757,10 +757,12 @@ Shape Tuplet::shape() const
 
 void Tuplet::scanElements(void* data, void (* func)(void*, Element*), bool all)
 {
-    if (_number && all) {
-        func(data, _number);
+    if (all || visible() || score()->showInvisible()) {
+        func(data, this);
+    }
+    if (all && _number) { // scan number only if all is true
+        Element::scanElements(data, func, all);
```

Comment:
Maybe have the number do this check rather than the tuplet?

Reply:
`_number` is of type `Text*` so, in general, Text should be scanned if it is visible, not just when `all` is true. That is why I cannot move this check to Text class.

Reply:
Makes sense. In that case you should probably do it like this:

```c++
for (child : this) {
    if (child == _number && !all) {
        continue; // don't scan number unless all is true
    }
    child->scanElements(data, func, all);
}
```

Reply:
okay, did it that way.